zebra-network/src/peer/connection.rs

```diff
@@ -127,34 +146,69 @@ impl Handler {
                     }
                 };
                 if items.iter().all(hash_in_hashes) {
-                    Handler::Finished(Err(PeerError::NotFound(items)))
+                    // all hashes have a transaction or a `NotFound` entry
+                    if !transactions.is_empty() {
+                        // TODO: does the caller need a list of missing transactions?
+                        Handler::Finished(Ok(Response::Transactions(transactions)))
+                    } else {
+                        Handler::Finished(Err(PeerError::NotFound(items)))
+                    }
                 } else {
+                    // waiting for more transactions or a complete `NotFound` message
+                    if items.iter().any(hash_in_hashes) {
+                        debug!(
+                            not_found_transactions = ?items.iter().cloned().filter(hash_in_hashes).count(),
+                            remaining_transactions = ?hashes.len(),
+                            "notfound containing some of the remaining requested transactions");
+                    }
                     ignored_msg = Some(Message::NotFound(items));
                     Handler::TransactionsByHash {
                         hashes,
                         transactions,
                     }
                 }
             }
+            // `zcashd` returns blocks sequentially (but not necessarily immediately).
+            // It silently skips missing blocks, rather than using `NotFound`:
```

Comment:
```suggestion
            // `zcashd` returns requested blocks in a single batch of messages.
            // Other blocks or non-blocks messages can come before or after the batch.
            // `zcashd` silently skips missing blocks, rather than sending a final `NotFound` message.
```